src/rules/key_value.rs

```diff
@@ -51,13 +51,17 @@ pub fn rule(
             if let rnix::SyntaxKind::NODE_ATTR_SET
             | rnix::SyntaxKind::NODE_LIST
             | rnix::SyntaxKind::NODE_PAREN
+            | rnix::SyntaxKind::NODE_WITH
+            | rnix::SyntaxKind::NODE_BIN_OP
```

Comment:
This bin-op here has the potential to look bad when multiline:

```diff
--- a/lib/customisation.nix
+++ b/lib/customisation.nix
@@ -169,22 +169,20 @@ rec {
   extendDerivation = condition: passthru: drv: let
     outputs = drv.outputs or [ "out" ];

-    commonAttrs =
-      drv
-      // (builtins.listToAttrs outputsList)
-      // ({ all = map (x: x.value) outputsList; })
-      // passthru;
+    commonAttrs = drv
+    // (builtins.listToAttrs outputsList)
+    // ({ all = map (x: x.value) outputsList; })
+    // passthru;
```

I would say that for simplicity, let's only change the `with` in this PR, and let's sort out the bin-op in another PR, additionally the diff seems to be smaller if we don't change the bin-op